### ISSUE TITLE
gitlab_branch_protection should ignore individual users

### ISSUE THREAD

**Issue by jeremad** [gitlabhq/terraform-provider-gitlab#441]
### Terraform Version
```
Terraform v0.12.29
+ provider.gitlab v2.11.0
```
### Affected Resource(s)
Please list the resources as a list, for example:
- gitlab_branch_protection

### Expected Behavior

Giving individual push access to protected branch is a premium feature of GitLab, the provider does not implement that yet.
If an individual user has push_access, the GitLab API returns:
```
"push_access_levels":[{
  "access_level": 40,
  "access_level_description": "Tanker Sysadmin",
  "user_id": 1706972,
  "group_id": null
},{
  "access_level": 0,
  "access_level_description": "No one",
  "user_id": null,
  "group_id": null
}]
```
And this is bugged: https://github.com/gitlabhq/terraform-provider-gitlab/blob/master/gitlab/resource_gitlab_branch_protection.go#L120

We should ignore push_access_levels and merge_access_level for individual users at least while the API is not implemented in the provider

**Comment by mattkasa** [comment]
I think this has been fixed by #556, but please feel free to reopen if I am mistaken 👍

**Comment by jeremad** [comment]
thx
